src/core/automount.c

```diff
@@ -925,14 +925,8 @@ static int automount_deserialize_item(Unit *u, const char *key, const char *valu
                                 log_unit_error_errno(u, r, "Failed to add expire token to set: %m");
                 }
         } else if (streq(key, "pipe-fd")) {
-                int fd;
-
-                if ((fd = parse_fd(value)) < 0 || !fdset_contains(fds, fd))
-                        log_unit_debug(u, "Failed to parse pipe-fd value: %s", value);
-                else {
-                        safe_close(a->pipe_fd);
-                        a->pipe_fd = fdset_remove(fds, fd);
-                }
+                safe_close(a->pipe_fd);
+                a->pipe_fd = deserialize_fd(fds, value);
```

Comment:
this is not exactly the same, beforehand pipe_fd remained valid if deserializing failed, now it will always be closed

Reply:
yeah, we are fucked anyway if serialization failed. This way we will at least not mix old and new config of an automount unit.

i mean, there really shouldn't be any "old" config anyway, we deserialize into an "empty" automount unit. You could as well drop the safe_close(), it's just there for paranoia...

Reply:
Sure, but if it's a justified change then it should be explained in the commit, including why it's ok to do - at the moment it just says it's modernizing, but that usually means mostly cosmetic changes

Reply:
(actually I mixed the two commits, it doesn't say it modernizes, but still it only says it adds a new helper to make it uniform)